Enterprise-CMCS/managed-care-review
# MCR-3849 Fix session timeout

## Summary
Address issues with session timeout and re-implement with [IdleTimer ](https://idletimer.dev/). 

**Issues addressed include:**
- logout and continue work across tabs
- if one tab has session timeout modal open then user opens another tab and then lets computer go to sleep, both tabs will be logged out of MC-Review after time elapses with no action
- if user tries to add a document with file upload but it fails due to IDM having been logged out in the background for whatever reason, MC-Review will also force logout at this point and display session timeout message so the user knows what happened

Still in progress
- [x] write up acceptance jams steps
- [x] finish adding `AuthenticatedRouteWrapper` tests for idle timer behaviors

#### Related issues
https://jiraent.cms.gov/browse/MCR-3849

#### Screenshots
![Screenshot 2024-09-04 at 4 13 22 PM](https://github.com/user-attachments/assets/73e399cf-540e-441f-af85-18ad00435057)

#### Test cases covered
Session Timeout acceptance jam document forthcoming

## QA guidance
Please re-test[ the steps from this comment in Jira](https://jiraent.cms.gov/browse/MCR-3849?focusedId=2925189&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2925189).

And also pick a couple of these [acceptance jam](https://docs.google.com/spreadsheets/d/1IYXcXkc47ogrjFWKrHUoBGFfNHrhfovXDU5CSf2oF0c/edit?gid=0#gid=0) cases to retest. Keep in mind the timeout for toph in review apps is set to 2min to make things go faster so you won't have to wait as long to verify things.
